primitives/src/parachain.rs

```diff
@@ -53,6 +54,56 @@ pub type ValidatorIndex = u32;
 /// so we define it to be the same type as `SessionKey`. In the future it may have different crypto.
 pub type ValidatorSignature = super::SessionSignature;
 
+/// Type determining the active set of parachains in current block.
+pub trait ActiveParas {
+	/// Return the active set of parachains in current block. This attempts to keep any IDs in the
+	/// same place between sequential blocks. It is therefore unordered. The second item in the
+	/// tuple is the required collator ID, if any. If `Some`, then it is invalid to include any
+	/// other collator's block.
+	///
+	/// NOTE: The initial implementation simply concatenates the (ordered) set of (permanent)
+	/// parachain IDs with the (unordered) set of parathread IDs selected for this block.
+	fn active_paras() -> Vec<(Id, Option<CollatorId>)>;
```

Comment:
```suggestion
	fn active_paras() -> Parachains;
```

suggest adding the type alias `pub type Parachains = Vec<(Id, Option<CollatorId>)>;` as well